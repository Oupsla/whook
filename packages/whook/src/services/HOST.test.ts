import _internalIp from 'internal-ip';
import initHOST from './HOST';

describe('initHOST', () => {
  const log = jest.fn();
  const internalIp = { v4: jest.fn() as jest.Mock & typeof _internalIp.v4 };

  beforeEach(() => {
    log.mockReset();
    internalIp.v4.mockReset();
  });

  it('should use the env HOST first', async () => {
    const HOST = await initHOST({
      ENV: { HOST: '192.168.1.11' },
      log,
      internalIp,
    });

    expect({
      HOST,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
      internalIpV4Calls: internalIp.v4.mock.calls,
    }).toMatchSnapshot();
  });

  it('should find a HOST by itself if no env HOST', async () => {
    internalIp.v4.mockResolvedValueOnce('192.168.1.10');

    const HOST = await initHOST({
      ENV: {},
      log,
      internalIp,
    });

    expect(HOST);
    expect({
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
      internalIpV4Calls: internalIp.v4.mock.calls,
    }).toMatchSnapshot();
  });

  it('should fallback to localhost', async () => {
    internalIp.v4.mockResolvedValueOnce('');

    const HOST = await initHOST({
      ENV: {},
      log,
      internalIp,
    });

    expect(HOST);
    expect({
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
      internalIpV4Calls: internalIp.v4.mock.calls,
    }).toMatchSnapshot();
  });
});
