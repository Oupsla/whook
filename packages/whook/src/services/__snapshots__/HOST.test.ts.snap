// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initHOST should fallback to localhost 1`] = `
Object {
  "internalIpV4Calls": Array [
    Array [],
  ],
  "logCalls": Array [
    Array [
      "debug",
      "üè≠ - Initializing the HOST service.",
    ],
    Array [
      "warning",
      "üö´ - Could not detect any host. Fallbacking to localhost.",
    ],
  ],
}
`;

exports[`initHOST should find a HOST by itself if no env HOST 1`] = `
Object {
  "internalIpV4Calls": Array [
    Array [],
  ],
  "logCalls": Array [
    Array [
      "debug",
      "üè≠ - Initializing the HOST service.",
    ],
    Array [
      "warning",
      "‚úî - Using detected host 192.168.1.10",
    ],
  ],
}
`;

exports[`initHOST should use the env HOST first 1`] = `
Object {
  "HOST": "192.168.1.11",
  "internalIpV4Calls": Array [],
  "logCalls": Array [
    Array [
      "debug",
      "üè≠ - Initializing the HOST service.",
    ],
    Array [
      "warning",
      "‚ôªÔ∏è - Using ENV host 192.168.1.11",
    ],
  ],
}
`;
